src/main/java/com/sucy/skill/cmd/CmdList.java

```diff
@@ -64,31 +72,86 @@ public void execute(ConfigurableCommand cmd, Plugin plugin, CommandSender sender
         // Disabled world
         if (sender instanceof Player && !SkillAPI.getSettings().isWorldEnabled(((Player) sender).getWorld()) && args.length == 0) {
             cmd.sendMessage(sender, DISABLED, "&4You cannot use this command in this world");
+            return;
+        }
+
+        // List loaded skills
+        if (args.length >= 1) {
+            String listType = args[0];
+            // Check Args
+            switch (listType) {
+                case "skills":
+                    if (!sender.hasPermission(Permissions.LIST_SKILLS)) {
+                        cmd.sendMessage(sender, NEEDS_ARGS, ChatColor.RED + "You do not have permission to list skills");
+                        return;
+                    }
+                    File file = new File(plugin.getDataFolder() + "/dynamic", "skills.yml");
+                    YamlConfiguration config = YamlConfiguration.loadConfiguration(file);
+                    Set<String> loadedSkills = config.getKeys(false);
+                    String loadedSkillsString = ChatColor.GOLD + "Loaded Skills: \n";
+                    Iterator<String> itr = loadedSkills.iterator();
+                    // Skip "loaded" key
+                    if (itr.hasNext()) itr.next();
+                    else {
+                        sender.sendMessage(loadedSkillsString + ChatColor.RED + "No skills loaded");
+                        return;
+                    }
+                    while (itr.hasNext()) {
+                        loadedSkillsString += ChatColor.GREEN + itr.next() + "\n";
+                    }
+                    sender.sendMessage(loadedSkillsString);
+                    return;
+                case "groups": 
+                    if (!sender.hasPermission(Permissions.LIST_GROUPS)) {
+                        cmd.sendMessage(sender, NEEDS_ARGS, ChatColor.RED + "You do not have permission to list groups");
+                        return;
+                    }
+                    File groupFile = new File(plugin.getDataFolder() + "/dynamic", "groups.yml");
+                    YamlConfiguration groupConfig = YamlConfiguration.loadConfiguration(groupFile);
+                    Set<String> loadedGroups = groupConfig.getKeys(false);
+                    String loadedGroupsString = ChatColor.GOLD + "Loaded Groups: \n";
+                    Iterator<String> itr2 = loadedGroups.iterator();
+                    // Skip "loaded" key
+                    if (itr2.hasNext()) itr2.next();
+                    else {
+                        sender.sendMessage(loadedGroupsString + ChatColor.RED + "No groups loaded");
+                        return;
+                    }
+                    while (itr2.hasNext()) {
+                        loadedGroupsString += ChatColor.GREEN + itr2.next() + "\n";
+                    }
+                    sender.sendMessage(loadedGroupsString);
```

Comment:
This is kind of over-engineered for what it needs to be? `SkillAPI.getSkills()` and `SkillAPI.getClasses()` would return the data you need for this, no?

Reply:
My bad, there's `getSkills`, `getGroups` and `getClasses`